cornflow-server/cornflow/endpoints/login.py

```diff
@@ -227,8 +241,83 @@ def auth_oid_authenticate(
         else:
             raise InvalidUsage("Invalid request")
 
+    def auth_external_authenticate(self, token: str = None, username: str = None, password: str = None):
```

Comment:
Why create a new type of authentication instead of doing OpenID right? I do not see why we need what you call external auth instead of having a proper OpenID authentication, which we didn't have.